partiql-value/src/lib.rs

```diff
@@ -23,13 +26,158 @@ pub enum Value {
     Boolean(bool),
     Integer(i64),
     Real(OrderedFloat<f64>),
+    Decimal(RustDecimal),
     String(Box<String>),
     Blob(Box<Vec<u8>>),
     List(Box<List>),
     Bag(Box<Bag>),
     Tuple(Box<Tuple>),
 }
 
+impl ops::Add for Value {
+    type Output = Self;
+
+    fn add(self, rhs: Self) -> Self::Output {
+        match (&self, &rhs) {
+            // TODO: edge cases dealing with overflow
+            (Value::Missing, _) => Value::Missing,
+            (_, Value::Missing) => Value::Missing,
+            (Value::Null, _) => Value::Null,
+            (_, Value::Null) => Value::Null,
+            (Value::Integer(l), Value::Integer(r)) => Value::Integer(l + r),
+            (Value::Real(l), Value::Real(r)) => Value::Real(*l + *r),
+            (Value::Decimal(l), Value::Decimal(r)) => Value::Decimal(l + r),
+            (Value::Integer(_), Value::Real(_)) => coerce_int_to_real(&self) + rhs,
+            (Value::Integer(_), Value::Decimal(_)) => coerce_int_or_real_to_decimal(&self) + rhs,
+            (Value::Real(_), Value::Decimal(_)) => coerce_int_or_real_to_decimal(&self) + rhs,
+            (Value::Real(_), Value::Integer(_)) => self + coerce_int_to_real(&rhs),
+            (Value::Decimal(_), Value::Integer(_)) => self + coerce_int_or_real_to_decimal(&rhs),
+            (Value::Decimal(_), Value::Real(_)) => self + coerce_int_or_real_to_decimal(&rhs),
```

Comment:
Not a blocker, perhaps for future: we could have a method for the BinOps to coerce the `lhs` in case of type mismatch before op overload; in that case we would end up w/ coercion only once.